src/DrWatson.jl

```diff
@@ -17,8 +17,13 @@ include("saving_tools.jl")
 # Functionality that saves/loads
 using FileIO: save, load
 export save, load
-wsave = save
-wload = load
+wsave(args...) = save(args...)
+function wsave(filename, obj)
+    mkpath(dirname(filename))
+    save(filename, obj)
+end
```

Comment:
Hm, maybe we can do this even better, so that `wsave` always calls `mkpath` regardless of arguments. At the moment if someone overloaded `wsave(::String, ::Plots.Plot, ...)` then they wouldn't get the automatic `mkpath` with it.

Hm, but this seems tricky... I've been thinking about it for two minutes but can't figure a simple way!

Reply:
Good point; could of course do something like
```julia
wsave(filename, obj) = (mkpath(dirname(filename)); _wsave(filename, obj))
```
and then have the user override `_wsave`. Not as nice-looking though :/

Reply:
Ah, I see your point. I didn't consider the scenario where someone would be using `wsave` directly, but rather only the higher level functions `safesave` or `tagsave`.

What is the use-case of using `wsave` directly, instead of e.g. `save`?

I am now also thinking about how would `tagsave` work with overloading `wsave` to save figures... How do you add git id to a figure?

Reply:
I think for the same reasons why we'd use it internally; it's more convenient to write `wsave` than `mkpath(...); save(filename, obj)`:) Also, maybe in the future more functionality is added to `wsave`?

More importantly, if the user wants to extend behavior for other functions using `wsave`, e.g. `tagsave`, the user ends up working with `wsave` instead of `_wsave`. 

But of course, this is all a matter of opinion. I just know that I like the approach taken by Distributions.jl with overloading `_logpdf` once and then you use `logpdf` everywhere else, knowing that certain checks are performed before passing to `_logpdf`. `wsave` is likely to see less frequent use compared to `logpdf` though:)

Reply:
> I am now also thinking about how would tagsave work with overloading wsave to save figures... How do you add git id to a figure?

This is basically the reason why I started looking into the internals of DrWatson:) I also briefly spoke to @JonasIsensee about this on Slack. I've looked at creating a project similar to DrWatson myself some time back, and I was planning on going for an approach where I have a simple `filename → metadata` lookup-table (started out as a simple json, but played with the  thought of eventually turning it into a SQLite table). You can then just add whatever information you want to any file, e.g. commit id.

Reply:
Other use-cases for this is when you have maybe have files too large to fit into memory, and so you want to read large files during whatever computation you're doing. Then it would be beneficial to have files in a parquet-format. You might still want to know which commit id produced it

Reply:
Okay, I see your point(s).

Okay then, go ahead and make the change with `_wsave` and we can merge this!

Reply:
related: https://github.com/JuliaDynamics/DrWatson.jl/issues/85

Reply:
Great! I'll do that asap:)